packages/rpc-core/src/rpc-methods/getMultipleAccounts.ts

```diff
@@ -0,0 +1,113 @@
+import { Base58EncodedAddress } from '@solana/addresses';
+
+import {
+    Base58EncodedDataResponse,
+    Base64EncodedDataResponse,
+    Base64EncodedZStdCompressedDataResponse,
+    Commitment,
+    DataSlice,
+    LamportsUnsafeBeyond2Pow53Minus1,
+    RpcResponse,
+    Slot,
+    U64UnsafeBeyond2Pow53Minus1,
+} from './common';
+
+type GetMultipleAccountsApiResponseBase = {
+    /** indicates if the account contains a program (and is strictly read-only) */
+    executable: boolean;
+    /** number of lamports assigned to this account */
+    lamports: LamportsUnsafeBeyond2Pow53Minus1;
+    /** pubkey of the program this account has been assigned to */
+    owner: Base58EncodedAddress;
+    /** the epoch at which this account will next owe rent */
+    rentEpoch: U64UnsafeBeyond2Pow53Minus1;
+    /** the data size of the account */
+    space: U64UnsafeBeyond2Pow53Minus1;
+} | null;
+
+type GetMultipleAccountsApiResponseWithBase58EncodedData_DEPRECATED = Readonly<{
+    data: Base58EncodedDataResponse;
+}>;
+
+type GetMultipleAccountsApiResponseWithBase64EncodedData = Readonly<{
+    data: Base64EncodedDataResponse;
+}>;
+
+type GetMultipleAccountsApiResponseWithBase64EncodedZStdCompressedData = Readonly<{
+    data: Base64EncodedZStdCompressedDataResponse;
+}>;
+
+type GetMultipleAccountsApiResponseWithJsonData = Readonly<{
+    data:
+        | Readonly<{
+              /** Name of the program that owns this account. */
+              program: string;
+              parsed: unknown;
+              space: U64UnsafeBeyond2Pow53Minus1;
+          }>
+        // If `jsonParsed` encoding is requested but a parser cannot be found for the given
+        // account the `data` field falls back to `base64`.
+        | Base64EncodedDataResponse;
+}>;
+
+type GetMultipleAccountsApiCommonConfig = Readonly<{
+    /** Defaults to `finalized` */
+    commitment?: Commitment;
+    /** The minimum slot that the request can be evaluated at */
+    minContextSlot?: Slot;
+}>;
```

Comment:
Maybe! I deliberately avoided trying to be clever with this PR because I noticed that `getProgramAccounts` introduces the exciting quirk that it can either return things nested in an RPC Response (like `getAccountInfo` and `getMultipleAccounts` do).. or not, depending on the inputs!

So once that is in I think it'd make sense to revisit this and see if there are actually common data structures that make sense to extract, or if the combined weirdness of the 3 RPC calls makes it easier to keep them separate